core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +376,174 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
+                  AuthorizationUtils.callAuthorizationPluginForMetadataObject(
+                      metalake,
+                      object,
+                      authorizationPlugin -> {
+                        authorizationPlugin.onRoleUpdated(
+                            roleEntity,
+                            RoleChange.updateSecurableObject(
+                                role, securableObject, newSecurableObject));
+                      });
+                }
+
+              } else {
+                updateSecurableObjects.add(securableObject);
+              }
+            }
+
+            // If the role doesn't contain the securable object, the role appends a new securable
+            // object.
+            if (!objectExist) {
+              SecurableObject securableObject =
+                  SecurableObjects.parse(
+                      object.fullName(), object.type(), Lists.newArrayList(privileges));
+              AuthorizationUtils.callAuthorizationPluginForMetadataObject(
+                  metalake,
+                  object,
+                  authorizationPlugin -> {
+                    authorizationPlugin.onRoleUpdated(
+                        roleEntity, RoleChange.addSecurableObject(role, securableObject));
+                  });
+              updateSecurableObjects.add(securableObject);
+            }
+
+            AuditInfo auditInfo =
+                AuditInfo.builder()
+                    .withCreator(roleEntity.auditInfo().creator())
+                    .withCreateTime(roleEntity.auditInfo().createTime())
+                    .withLastModifier(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withLastModifiedTime(Instant.now())
+                    .build();
+
+            return RoleEntity.builder()
+                .withId(roleEntity.id())
+                .withName(roleEntity.name())
+                .withNamespace(roleEntity.namespace())
+                .withProperties(roleEntity.properties())
+                .withAuditInfo(auditInfo)
+                .withSecurableObjects(updateSecurableObjects)
+                .build();
+          });
+    } catch (NoSuchEntityException nse) {
+      LOG.warn("Failed to grant, role {} does not exist in the metalake {}", role, metalake, nse);
+      throw new NoSuchRoleException(ROLE_DOES_NOT_EXIST_MSG, role, metalake);
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  Role revokePrivilegeFromRole(
+      String metalake, String role, MetadataObject object, List<Privilege> privileges) {
+    try {
+      return store.update(
+          AuthorizationUtils.ofRole(metalake, role),
+          RoleEntity.class,
+          Entity.EntityType.ROLE,
+          roleEntity -> {
+            List<SecurableObject> securableObjects = roleEntity.securableObjects();
+
+            List<SecurableObject> updateSecurableObjects = Lists.newArrayList();
+            // Remove all the matching securable object privileges
```

Comment:
What if all the securable object are not matched, or the privileges are not/partially matched, should you at least add some logs?

Reply:
Added.